src/handlers/deeplinks.ts

```diff
@@ -337,3 +354,71 @@ function handleWalletConnect(uri?: string, connector?: string) {
     store.dispatch(walletConnectSetPendingRedirect());
   }
 }
+
+const isValidAsset = (s: string | undefined): s is UniqueId => {
+  if (!s) return false;
+  const [address, chainId] = s.split('_');
+  return supportedSwapChainIds.includes(parseInt(chainId, 10)) && (address === 'eth' || address.length === 42);
+};
+
+const querySearchAsset = async (uniqueId: string | undefined): Promise<ParsedSearchAsset | undefined> => {
+  if (!isValidAsset(uniqueId)) return;
+
+  const { address, chainId } = deriveAddressAndChainWithUniqueId(uniqueId);
+
+  const userAsset = queryUserAssets().then(allUserAssets => allUserAssets[chainId][uniqueId]);
+
+  const searchAsset = await queryTokenSearch({
+    chainId,
+    query: address.toLowerCase(),
+    keys: ['address'],
+    threshold: 'CASE_SENSITIVE_EQUAL',
+    list: 'verifiedAssets',
+  }).then(res => res[0]);
+
+  // @ts-expect-error aaaa
+  if (!searchAsset) return userAsset;
+
+  return parseSearchAsset({ searchAsset, userAsset: await userAsset });
+};
```

Comment:
I think a lot of this can be done in parallel so we can speed up this quite a bit.